### PR TITLE
PARQUET-2414: Add test file for additional BYTE_STREAM_SPLIT types

### PR DESCRIPTION
**Pull request by pitrou**
Add a new data file that allows exercising BYTE_STREAM_SPLIT for all supported types: FLOAT, DOUBLE, INT32, INT64, FIXED_LEN_BYTE_ARRAY (the latter with several widths and logical types).

For each type, two columns are provided with the same values: one PLAIN-encoded, the other BYTE_STREAM_SPLIT-encoded.

### CODE REVIEW

**Comment by pitrou** [comment]
This should not be merged until PARQUET-2414 is voted on.

File: data/README.md [hunk 1]
```diff
@@ -351,3 +353,37 @@ pq.write_table(
...
 This is a practical case where `BYTE_STREAM_SPLIT` encoding obtains a smaller file size than `PLAIN` or dictionary.
 Since the distributions are random normals centered at 0, each byte has nontrivial behavior.
+
+# Additional types
+
+`byte_stream_split_extended.gzip.parquet` is generated by pyarrow 16.0.0.
+It contains 7 pairs of columns, each in two variants containing the same
+values: one `PLAIN`-encoded and one `BYTE_STREAM_SPLIT`-encoded:
+```
+Version: 2.6
```
Comment (wgtmac) [inline data/README.md:363]: hmm... it seems we need to fix the version so here can correctly displays version 2.11

**Review by wgtmac — APPROVED** [review]

File: data/README.md (hunk 1 above)
Comment (pitrou) [reply data/README.md:363]: That will have to be part of the final C++ PR. We do not need the C++ changes to fix this README if we want to, though.

Comment (pitrou) [reply data/README.md:363]: (also see https://github.com/apache/arrow/issues/40096)

**Pull request merged**